packages/camera/camera_windows/example/lib/main.dart

```diff
@@ -0,0 +1,417 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:camera_platform_interface/camera_platform_interface.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+/// Example app for Camera Windows plugin.
+class MyApp extends StatefulWidget {
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  String _cameraInfo = 'Unknown';
+  List<CameraDescription> _cameras = <CameraDescription>[];
+  int _cameraIndex = 0;
+  int _cameraId = -1;
+  bool _initialized = false;
+  bool _recording = false;
+  bool _recordingTimed = false;
+  bool _recordAudio = true;
+  bool _previewPaused = false;
+  Size? _previewSize;
+  ResolutionPreset _resolutionPreset = ResolutionPreset.veryHigh;
+  StreamSubscription<CameraErrorEvent>? _errorStreamSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsFlutterBinding.ensureInitialized();
+    _getAvailableCameras();
+  }
+
+  @override
+  void dispose() {
+    _disposeCurrentCamera();
+    _errorStreamSubscription?.cancel();
+    _errorStreamSubscription = null;
+    super.dispose();
+  }
+
+  /// Fetches list of available cameras from camera_windows plugin.
+  Future<void> _getAvailableCameras() async {
+    String cameraInfo;
+    List<CameraDescription> cameras = <CameraDescription>[];
+
+    int cameraIndex = _cameraIndex;
+    try {
+      cameras = await CameraPlatform.instance.availableCameras();
+      cameraIndex = cameraIndex % cameras.length;
+      if (cameras.isEmpty) {
```

Comment:
In the case where `cameras.isEmpty` is true, the modulo operation on the line above will throw `IntegerDivisionByZeroException`. Consider doing that operation conditionally in the `else` block of the conditional. 

In the `if` block you could reset to some obviously invalid index. -1 would make it incredibly obvious that the index is invalid, but you'd need to account for it in the modulo expression in the else, since you'd (probably?) want to reset to index 0 when you finally get a list of cameras.

Since technically every index is invalid, 0 seems like an equally fair choice, and behaves nicely with the modulo call the next time this is called.